build.rs

```diff
@@ -518,13 +516,15 @@ fn windows_hardcoded_cross_compile(
 fn load_cross_compile_info(
     cross_compile_config: CrossCompileConfig,
 ) -> Result<(InterpreterConfig, BuildFlags)> {
-    let target_family = env::var("CARGO_CFG_TARGET_FAMILY")?;
-    // Because compiling for windows on linux still includes the unix target family
-    if target_family == "unix" {
+    match env::var("CARGO_CFG_TARGET_FAMILY") {
         // Configure for unix platforms using the sysconfigdata file
-        load_cross_compile_from_sysconfigdata(cross_compile_config)
-    } else {
-        windows_hardcoded_cross_compile(cross_compile_config)
+        Ok(os) if os == "unix" => load_cross_compile_from_sysconfigdata(cross_compile_config),
+        // Use hardcoded interpreter config when targeting Windows
+        Ok(os) if os == "windows" => windows_hardcoded_cross_compile(cross_compile_config),
+        // ???
+        Ok(os) => Err(format!("Unsupported target OS family: {}", os).into()),
+        // wasm32-wasi / bare metal ?
+        Err(_) => Err("CARGO_CFG_TARGET_FAMILY is not set when cross-compiling".into()),
```

Comment:
With `var_os`:

```suggestion
        Some(os) if os == "unix" | None => load_cross_compile_from_sysconfigdata(cross_compile_config),
        // Use hardcoded interpreter config when targeting Windows
        Some(os) if os == "windows" => windows_hardcoded_cross_compile(cross_compile_config),
        // ???
        Some(os) => bail!("Unsupported target OS family for cross-compilation: {:?}", os),
```

I also moved the "missing" case to just load sysconfigdata - I don't see any harm in trying to load that, and it's probably better than just giving up.